Source/Scene/Cesium3DTileStyleEngine.js

```diff
@@ -74,36 +76,40 @@ define([
                 //   2) this tile is now visible, but it wasn't visible when the style was first assigned
                 if (tile.lastStyleTime !== lastStyleTime) {
                     tile.lastStyleTime = lastStyleTime;
-                    styleCompositeContent(this, tile.content, stats);
+                    styleCompositeContent(this, frameState, tile.content, stats);
 
                     ++stats.numberOfTilesStyled;
                 }
             }
         }
     };
 
-    function styleCompositeContent(styleEngine, content, stats) {
+    function styleCompositeContent(styleEngine, frameState, content, stats) {
         var innerContents = content.innerContents;
         if (defined(innerContents)) {
             var length = innerContents.length;
             for (var i = 0; i < length; ++i) {
                 // Recurse for composites of composites
-                styleCompositeContent(styleEngine, innerContents[i], stats);
+                styleCompositeContent(styleEngine, frameState, innerContents[i], stats);
             }
         } else {
             // Not a composite tile
-            styleContent(styleEngine, content, stats);
+            styleContent(styleEngine, frameState, content, stats);
         }
     }
 
     var scratchColor = new Color();
 
-    function styleContent(styleEngine, content, stats) {
+    function styleContent(styleEngine, frameState, content, stats) {
         var length = content.featuresLength;
         var style = styleEngine._style;
 
         stats.numberOfFeaturesStyled += length;
 
+        if (content.applyStyleWithShader(frameState, style)) {
```

Comment:
No because `clearStyle` below updates the batch table features only. `applyStyleWithShader` will handle clearing the shader style when `style` is undefined.